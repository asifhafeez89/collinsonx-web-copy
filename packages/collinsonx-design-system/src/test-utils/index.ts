import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
