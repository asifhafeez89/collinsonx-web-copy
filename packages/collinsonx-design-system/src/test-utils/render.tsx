import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
