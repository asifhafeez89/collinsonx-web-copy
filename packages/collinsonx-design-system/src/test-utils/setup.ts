import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
