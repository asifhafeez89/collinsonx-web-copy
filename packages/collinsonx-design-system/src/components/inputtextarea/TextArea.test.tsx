import renderer from 'react-test-renderer';
import TextArea from '.';
import { render, screen, waitFor, Provider } from 'test-utils';
import userEvent from '@testing-library/user-event';

const mockFn = jest.fn();

describe('<Textarea />', () => {
  it('renders Textarea', () => {
    const tree = renderer
      .create(
        <Provider>
          <TextArea
            placeholder="Your comment"
            label="Your comment"
            withAsterisk
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can type ', async () => {
    render(
      <>
        <TextArea
          placeholder="Your comment"
          label="Your comment"
          withAsterisk
          onChange={mockFn}
        />
      </>
    );

    const inputnode = screen.getByPlaceholderText('Your comment');

    userEvent.type(inputnode, 'Hello');

    await waitFor(() => {
      expect(mockFn).toHaveBeenCalledTimes(5);
    });
  });
});
