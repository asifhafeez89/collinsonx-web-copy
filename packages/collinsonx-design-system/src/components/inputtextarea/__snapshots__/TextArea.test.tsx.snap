// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> renders Textarea 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}",
      }
    }
    data-mantine-styles="classes"
  />,
  <div
    className="m-46b77525 mantine-InputWrapper-root mantine-Textarea-root"
    style={
      {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <label
      className="m-8fdc1311 mantine-InputWrapper-label mantine-Textarea-label"
      data-required={true}
      htmlFor="mantine-"
      id="mantine--label"
      onMouseDown={[Function]}
      style={
        {
          "color": "#000000",
          "order": -2,
        }
      }
    >
      Your comment
      <span
        aria-hidden={true}
        className="m-78a94662 mantine-InputWrapper-required mantine-Textarea-required"
        style={{}}
      >
         *
      </span>
    </label>
    <div
      className="m-6c018570 mantine-Input-wrapper mantine-Textarea-wrapper"
      data-multiline={true}
      data-variant="default"
      style={{}}
    >
      <textarea
        aria-invalid={false}
        className="m-8fb7ebe7 mantine-Input-input mantine-Textarea-input"
        data-variant="default"
        id="mantine-"
        placeholder="Your comment"
        style={
          {
            "order": -1,
          }
        }
      />
    </div>
  </div>,
]
`;
